File: backend/internal/domain/note.go
package domain

import "time"

// Note は学習メモ。IsPinned と IsPublic は独立した属性。
type Note struct {
	ID        uint64    `json:"id"`
	UserID    uint64    `json:"userId"`
	Title     string    `json:"title"`
	Content   string    `json:"content"`
	IsPublic  bool      `json:"isPublic"`
	IsPinned  bool      `json:"isPinned"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

File: backend-java/src/main/java/com/normanblog/frestyle/entity/Note.java
package com.normanblog.frestyle.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/** ユーザーの学習ノートを表す JPA エンティティ。 */
@Entity
@Table(name = "notes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Note {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "user_id")
  private Long userId;

  @Column(nullable = false)
  private String title;

  @Column(columnDefinition = "text")
  private String content;

  @Column(name = "is_public")
  private boolean isPublic;

  @Column(name = "is_pinned")
  private boolean isPinned;

  @Column(name = "created_at")
  private Instant createdAt;

  @Column(name = "updated_at")
  private Instant updatedAt;
}
